src/Serializer/AbstractItemNormalizer.php

```diff
@@ -899,4 +877,28 @@ private function setValue(object $object, string $attributeName, mixed $value):
             // Properties not found are ignored
         }
     }
+
+    private function createOperationContext(array $context, ?string $resourceClass = null): array {
+        if (isset($context['operation'])) {
+            $context['previous_operation'] = $context['operation'];
+            $context['previous_operation_name'] = $context['operation_name'];
+
+            if (!isset($context['root_operation'])) {
+                $context['root_operation'] = $context['operation'];
+                $context['root_operation_name'] = $context['operation_name'];
+            }
+        }
+
+        unset($context['iri'], $context['uri_variables']);
+        if ($resourceClass) {
+            unset($context['operation'], $context['operation_name']);
+            $context['resource_class'] = $resourceClass;
+            if ($this->resourceMetadataCollectionFactory) {
+                $context['operation'] = $this->resourceMetadataCollectionFactory->create($resourceClass)->getOperation();
+                $context['operation_name'] = $context['operation']?->getName();
+            }
+        }
+
+        return $context;
+    }
```

Comment:
@jonag can you try this and use `root_operation`?

Reply:
Updating to API Platform v3.1.14 and changing my custom normalizer to use the root_operation instead of the operation fixes all my tests! 
Thank you 👏!